File: backend/src/main/java/top/vuhe/common/exception/SystemProcessingException.java
package top.vuhe.common.exception;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 系统运行异常
 *
 * @author zhuhe
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class SystemProcessingException extends RuntimeException {
    private int code;
    private String msg;

    public SystemProcessingException() {
        super(ExceptionEnum.UNKNOWN.getMessage());
        this.code = 500;
        this.msg = ExceptionEnum.UNKNOWN.getMessage();
    }

    public SystemProcessingException(ExceptionEnum eEnum) {
        this.code = eEnum.getCode();
        this.msg = eEnum.getMessage();
    }

    public SystemProcessingException(String e) {
        this.code = 500;
        this.msg = e;
    }

    public SystemProcessingException(ExceptionEnum eEnum, Throwable e) {
        super(eEnum.getMessage(), e);
        this.code = eEnum.getCode();
        this.msg = eEnum.getMessage();
    }

}

File: backend-kt/src/main/kotlin/top/vuhe/sw/common/exception/SystemProcessingException.kt
package top.vuhe.sw.common.exception

import java.lang.RuntimeException

/**
 * 系统运行异常
 *
 * @author zhuhe
 */
class SystemProcessingException : RuntimeException {
    val msg: String
    val code: Int

    constructor() : super(ExceptionEnum.UNKNOWN.message) {
        this.code = 500
        this.msg = ExceptionEnum.UNKNOWN.message
    }

    constructor(eEnum: ExceptionEnum) {
        this.code = eEnum.code
        this.msg = eEnum.message
    }

    constructor(e: String) {
        this.code = 500
        this.msg = e
    }

    constructor(eEnum: ExceptionEnum, e: Throwable) : super(eEnum.message, e) {
        this.code = eEnum.code
        this.msg = eEnum.message
    }

}